packages/protocol/contracts/governance/Election.sol

```diff
@@ -845,4 +885,40 @@ contract Election is
     }
     return res;
   }
+
+  struct SlashVotesInfo {
+    address[] groups;
+    uint256 remainingValue;
+  }
+
+  // Iterates in reverse order over all groups voted for by `account`,
+  // reducing the total amount of voting gold by `value` in total.
+  // For each group, if there `value` gold has left to be slashed, reduces
+  // the number of pending and active votes for that group by as much as
+  // possible without exceeding the total value that needs to be slashed.
+  function slashVotes(
+    address account,
+    uint256 value,
+    address[] calldata lessers,
+    address[] calldata greaters,
+    uint256[] calldata indices
+  ) external nonReentrant returns (bool) {
+    SlashVotesInfo memory info = SlashVotesInfo(votes.groupsVotedFor[account], value);
+    for (uint256 i = info.groups.length; i > 0; i = i.sub(1)) {
```

Comment:
nit: Include a comment with the rationale for iterating backwards

Reply:
From your docs on slashing V3, `Iterates in reverse order over all groups voted for by account` - I don't have a great reason for the reverse ordering. I assumed it was in order to slash the most recently voted for groups first, since maybe they are more likely to be inactive. Is there a deeper reason? Or should I just switch the order?

Reply:
I think it makes sense, groups that were voted for first are more likely to be more highly preferred by the voter

Reply:
The way arrar deletes are implement (switches idx to delete with last idx & decrement arr.size) the above is not actually true. Array orders doesn't really mean "voted first".

So, not sure if this is worth the trouble